flake.nix

```diff
@@ -59,34 +59,52 @@
-                base
-                (pkgs.writeTextFile {
-                  name = base.name + "-activate-path";
-                  text = ''
-                    #!${pkgs.runtimeShell}
-                    set -euo pipefail
-
-                    ${activate}
-                  '';
-                  executable = true;
-                  destination = "/deploy-rs-activate";
-                })
-                (pkgs.writeTextFile {
-                  name = base.name + "-activate-rs";
-                  text = ''
-                    #!${pkgs.runtimeShell}
-                    exec ${self.defaultPackage."${system}"}/bin/activate "$@"
-                  '';
-                  executable = true;
-                  destination = "/activate-rs";
-                })
-              ];
-            };
-
-            nixos = base: custom base.config.system.build.toplevel ''
+            custom =
+              {
+                __toString = customSelf: "TODO: dryActivate";
+                __functor = customSelf: base: activate:
+                  pkgs.buildEnv {
+                    name = ("activatable-" + base.name);
+                    paths =
+                      let
+                        hasDryActivate = builtins.hasAttr "dryActivate" customSelf;
+                      in [
+                        base
+                        (pkgs.writeTextFile {
+                          name = base.name + "-activate-path";
+                          text = ''
+                            #!${pkgs.runtimeShell}
+                            set -euo pipefail
+
+                            if [[ $DRY_ACTIVATE == "1" ]]
+                            then
+                                if ${pkgs.lib.boolToString hasDryActivate}
+                                then
+                                    ${if hasDryActivate then customSelf.dryActivate else ":"}
+                                else
+                                    echo ${pkgs.writeScript "activate" activate}
+                                fi
+                            else
+                                ${activate}
+                            fi
+                          '';
+                          executable = true;
+                          destination = "/deploy-rs-activate";
+                        })
+                        (pkgs.writeTextFile {
+                            name = base.name + "-activate-rs";
+                            text = ''
+                            #!${pkgs.runtimeShell}
+                            exec ${self.defaultPackage."${system}"}/bin/activate "$@"
+                          '';
+                          executable = true;
+                          destination = "/activate-rs";
+                        })
+                      ];
+                  };
+              };
+
+            nixos = base: (custom // { dryActivate = "$PROFILE/bin/switch-to-configuration dry-activate"; }) base.config.system.build.toplevel ''
```

Comment:
This is not quite the interface I had in mind, but ok :) I was thinking more along the lines of just being able to `custom profile activation-script dry-activation script`

This should be doable with a `__functor//__toString` set but if you don't want to, this interface is good enough!

Reply:
I think attrsset instead of positional arguments is better in case we ever want to augment it futher in the future

Reply:
@balsoft I don't understand your propose well enough to apply it, sorry.

@notgne2 I applied all your suggestions.